include/oneapi/dpl/pstl/hetero/dpcpp/parallel_backend_sycl_radix_sort.h

```diff
@@ -681,13 +644,22 @@ __radix_sort_reorder_submit(_ExecutionPolicy&& __exec, ::std::size_t __segments,
                 // 1. create a private array for storing offset values
                 //    and add total offset and offset for compute unit for a certain radix state
                 _OffsetT __offset_arr[__radix_states];
+                _OffsetT __scanned_bin = 0;
+                const ::std::size_t __scan_size = __segments + 1;
                 const ::std::uint32_t __global_offset_start_idx = (__segments + 1) * __radix_states;
                 for (::std::uint32_t __radix_state_idx = 0; __radix_state_idx < __radix_states; ++__radix_state_idx)
                 {
                     const ::std::uint32_t __global_offset_idx = __global_offset_start_idx + __radix_state_idx;
                     const ::std::uint32_t __local_offset_idx = __wgroup_idx + (__segments + 1) * __radix_state_idx;
-                    __offset_arr[__radix_state_idx] =
-                        __offset_rng[__global_offset_idx] + __offset_rng[__local_offset_idx];
+
+                    //scan bins (serial)
+                    if (__radix_state_idx > 0)
+                    {
+                        ::std::size_t __last_segment_bucket_idx = __radix_state_idx * __scan_size - 1;
+                        __scanned_bin += __offset_rng[__last_segment_bucket_idx];
+                    }
+
+                    __offset_arr[__radix_state_idx] = __scanned_bin + __offset_rng[__local_offset_idx];
```

Comment:
I believe this change would be be mutually exclusive with #680.

In #680, we changed `__offset_arr` to be in local memory instead of private per work item. We also changed this loop to cooperatively fill the array so that each work item does not need to iterate over the entire space of radix states. This reduces both the space requirement and compute time, which is noticeable when we increase the number of radix bits from 4 to 6.

Reply:
So, your changes  should be merged firstly, ok. 
BTW, what's about the second kernel for the scan phase? Did you removed one as well?

Reply:
No, I didn't modify either of the scan kernels.

Reply:
Adam, are you fine if we commit this patch and resolve the conflict in #680 if/when we proceed with it?

Reply:
Yes, that works for me. I'm in the process of splitting #680 anyway so that it won't conflict with this PR.